internal/controller/csiaddons/persistentvolumeclaim_controller.go

```diff
@@ -114,6 +118,11 @@ func (r *PersistentVolumeClaimReconciler) Reconcile(ctx context.Context, req ctr
 		return ctrl.Result{}, nil
 	}
 
+	err = r.processKeyRotation(ctx, &logger, &req, pvc, pv)
```

Comment:
If keyRotation errors out for some reason, then ReclaimSpace will not be processed?

Reply:
Correct, one of the side effects of having a single pv controller for both the operations :( We could change the ordering but then the situation would be reversed, if relciamspace fails, encryption key rotation would not be processed. 

Using sync would not benefit as if the current reconcile returns (from any sub-routine), the reconcile will end.

One possible solution would be to implement a new storage class controller for key rotation but then a SC can have a lot of PVCs and we will have to process each one of them, a very significant overhead. This was discussing during the design phase.

Reply:
can we keep this a two different operations like run in parallel and reqeue if one fails?

Reply:
Better yet, let's just log the encryption key rotation if it fails and let reclaim space continue. In the end we check the return values for both and requeue if any of them errored out.

Reply:
This is implemented now, PTAL @Madhu-1 @iPraveenParihar